metrics/details/init.lua

```diff
@@ -136,8 +140,14 @@ end
 local Counter = {}
 Counter.__index = Counter
 
-function Counter.new(name, help)
+function Counter.new(name, help, opts)
+    local opts = opts or {}
+
     local obj = Shared.new(name, help, 'counter')
+    if not opts.dont_register then
```

Comment:
I don't like `dont_regiser` name.
Maybe it's better to name this opt `not_register` or `hide`?
Or maybe use `register` name and check `if opts.register == false then ...`?

BTW, why only `Counter` has this option?
I think the opportunity to create not registered collectors would be useful.

Reply:
`dont_register` means don't register.
As long is it's clear what it does and is not too long -- I consider naming a success :)

Only `Counter` has the option because only counter participate in Histogram collector.
Notice, that the flag is not exposed to a public API.